src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -931,3 +824,173 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
+    kubernetes_version = get_server_version(configuration)
+
+    # Checking helm installation
+    check_helm_install(kube_config, kube_context)
+
+    # Check helm version
+    helm_version = check_helm_version(kube_config, kube_context)
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.HelmVersion': helm_version})
+
+    # Check for faulty pre-release helm versions
+    if "3.3.0-rc" in helm_version:
+        telemetry.set_user_fault()
+        raise CLIError("The current helm version is not supported for azure-arc onboarding. Please upgrade helm to a stable version and try again.")
+
+    # Check Release Existance
+    release_namespace = get_release_namespace(kube_config, kube_context)
+    if release_namespace:
+        # Loading config map
+        api_instance = kube_client.CoreV1Api(kube_client.ApiClient(configuration))
+        try:
+            configmap = api_instance.read_namespaced_config_map('azure-clusterconfig', 'azure-arc')
+        except Exception as e:  # pylint: disable=broad-except
+            utils.kubernetes_exception_handler(e, consts.Read_ConfigMap_Fault_Type, 'Unable to read ConfigMap',
+                                               error_message="Unable to read ConfigMap 'azure-clusterconfig' in 'azure-arc' namespace: ",
+                                               message_for_not_found="The helm release 'azure-arc' is present but the azure-arc namespace/configmap is missing. Please run 'helm delete azure-arc --no-hooks' to cleanup the release before onboarding the cluster again.")
+
+        auto_update_enabled = configmap.data["AZURE_ARC_AUTOUPDATE"]
+        if auto_update_enabled == "true":
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception='connectedk8s upgrade called when auto-update is set to true', fault_type=consts.Manual_Upgrade_Called_In_Auto_Update_Enabled,
+                                    summary='az connectedk8s upgrade to manually upgrade agents and extensions is only supported when auto-upgrade is set to false.')
+            raise CLIError("az connectedk8s upgrade to manually upgrade agents and extensions is only supported when auto-upgrade is set to false. Please run az connectedk8s update -n <> -g <> --auto-upgrade 'false' before performing manual upgrade")
+
+    else:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception="The azure-arc release namespace couldn't be retrieved", fault_type=consts.Release_Namespace_Not_Found,
+                                summary="The azure-arc release namespace couldn't be retrieved, which implies that the kubernetes cluster has not been onboarded to azure-arc.")
+        raise CLIError("The azure-arc release namespace couldn't be retrieved, which implies that the kubernetes cluster has not been onboarded to azure-arc.")
+
+        # Check whether Connected Cluster is present
+    if not connected_cluster_exists(client, resource_group_name, cluster_name):
```

Comment:
We should validate if the cluster name and rg name provided by user are the same values stored in config map. If the provided values point to different ARM resource, it would be a problem. (We do this in connectedk8s connect)

Reply:
Updated